MembraneRTC/Sources/MembraneRTC/MembraneRTC.swift

```diff
@@ -338,20 +364,93 @@ public class MembraneRTC: MulticastDelegate<MembraneRTCDelegate>, ObservableObje
             self.config.iceServers = [Self.defaultIceServer()]
         }
 
-        guard let peerConnection = ConnectionManager.createPeerConnection(config, constraints: Self.mediaConstraints) else {
+        guard let peerConnection = connectionManager.createPeerConnection(config, constraints: Self.mediaConstraints) else {
             fatalError("Failed to initialize new PeerConnection")
         }
         connection = peerConnection
 
         peerConnection.delegate = self
         
         localTracks.forEach { track in
-            peerConnection.add(track.rtcTrack(), streamIds: [localStreamId])
+            if(track.rtcTrack().kind == "video" && (track as? LocalVideoTrack)?.simulcastConfig.enabled == true) {
+                let simulcastConfig = (track as? LocalVideoTrack)?.simulcastConfig
+                
+                let sendEncodings = [
+                    RTCRtpEncodingParameters.create(rid: "l", active: false, scaleResolutionDownBy:4.0),
+                    RTCRtpEncodingParameters.create(rid: "m", active: false, scaleResolutionDownBy:2.0),
+                    RTCRtpEncodingParameters.create(rid: "h", active: false, scaleResolutionDownBy:1.0),
+                ]
+                
+                simulcastConfig?.activeEncodings.forEach { enconding in
+                    sendEncodings[enconding.rawValue].isActive = true;
+                }
+
+                let transceiverInit = RTCRtpTransceiverInit()
+                transceiverInit.direction = RTCRtpTransceiverDirection.sendOnly
+                transceiverInit.sendEncodings = sendEncodings
+                transceiverInit.streamIds = [localStreamId]
+
+                peerConnection.addTransceiver(with: track.rtcTrack(), init: transceiverInit)
+            } else {
+                peerConnection.add(track.rtcTrack(), streamIds: [localStreamId])
+            }
             
         }
         
         peerConnection.enforceSendOnlyDirection()
     }
+
+    
+    /**
+      Selects track encoding that server should send to the client library.
+     
+         - Parameters:
+            - peerId:an id of a peer that owns the track
+            - trackId: an id of a remote track
+            - encoding: an encoding to receive
+     */
+    public func selectTrackEncoding(peerId: String, trackId: String, encoding: TrackEncoding) {
+        transport.send(event: SelectEncodingEvent(peerId: peerId, trackId: trackId, encoding: encoding.description))
+    }
+    
+    private func setTrackEncoding(trackId: String, encoding: TrackEncoding, enabled: Bool) {
```

Comment:
I am not sure if this is conformant with Swift style but if yes, then I would order methods in a way that public ones are at the beginning and private ones are after them